aws/data_source_aws_launch_template.go

```diff
@@ -349,29 +349,10 @@ func dataSourceAwsLaunchTemplateRead(d *schema.ResourceData, meta interface{}) e
 		LaunchTemplateNames: []*string{aws.String(d.Get("name").(string))},
 	})
 
-	if isAWSErr(err, ec2.LaunchTemplateErrorCodeLaunchTemplateIdDoesNotExist, "") {
-		log.Printf("[WARN] launch template (%s) not found - removing from state", d.Id())
-		d.SetId("")
-		return nil
-	}
-
-	// AWS SDK constant above is currently incorrect
-	if isAWSErr(err, "InvalidLaunchTemplateId.NotFound", "") {
-		log.Printf("[WARN] launch template (%s) not found - removing from state", d.Id())
-		d.SetId("")
-		return nil
-	}
-
 	if err != nil {
 		return fmt.Errorf("Error getting launch template: %s", err)
 	}
 
-	if dlt == nil || len(dlt.LaunchTemplates) == 0 {
-		log.Printf("[WARN] launch template (%s) not found - removing from state", d.Id())
-		d.SetId("")
-		return nil
-	}
-
```

Comment:
generally we shouldn't expect nil responses but just to err on the side of caution with any unexpected API responses, let's keep this check but since we want to return an error when 0 results are found for this data-source, lets adjust the block with something like:
```go
if dlt == nil || len(dlt.LaunchTemplates) == 0 {
    return errors.New("error reading launch template: empty output")
}
```

Reply:
in addition, since this is a singular data-source, we should also ensure that the # of launch templates we intend to find is only 1 and if more than 1 is returned we should notify the user with an error. this is a bit out of scope of this PR but as we're looking to update data sources that fall within this issue #13410, it would be great to have coverage added here e.g.
```go
if len(dlt.LaunchTemplates) > 1 {
		return fmt.Errorf("Search returned %d result(s), please revise so only one is returned", len(dlt.LaunchTemplates))
	}
```